src/Languages/Editor/Impl/Settings/EditorSettings.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Microsoft.Languages.Core.Formatting;
+
+namespace Microsoft.Languages.Editor.Settings {
+    public abstract class EditorSettings : IWritableEditorSettings, IEditorSettings {
+        public const string AutoFormatKey = "AutoFormat";
+        public const string CompletionEnabledKey = "CompletionEnabled";
+        public const string FormatterIndentSizeKey = "FormatterIndentSize";
+        public const string FormatterTabSizeKey = "FormatterTabSize";
+        public const string FormatterIndentTypeKey = "FormatterIndentType";
+        public const string IndentStyleKey = "IndentStyle";
+        public const string SyntaxCheckKey = "SyntaxCheckEnabled";
+        public const string InsertMatchingBracesKey = "InsertMatchingBraces";
+        public const string SignatureHelpEnabledKey = "SignatureHelpEnabled";
+
+        protected IEditorSettingsStorage Storage { get; }
+        protected IWritableEditorSettingsStorage WritableStorage { get; }
+
+        public EditorSettings(IEditorSettingsStorage storage) {
+            Storage = storage;
+            WritableStorage = storage as IWritableEditorSettingsStorage;
+            Storage.SettingsChanged += (s, e) => SettingsChanged?.Invoke(this, EventArgs.Empty);
```

Comment:
Event handler chain is usually an anti-pattern. If SettingsChanged is just a proxy, it would be better to change it to `add/remove`:
```cs
        public event EventHandler<EventArgs> SettingsChanged {
            add => Storage.SettingsChanged += value;
            remove => Storage.SettingsChanged -= value;
        }
```